app/services/organizations/csv_import_service.rb

```diff
@@ -0,0 +1,37 @@
+require "csv"
+
+module Organizations
+  class CsvImportService
+    def initialize(file, organization)
+      @file = file
+      @organization = organization
+    end
+
+    def call
+      CSV.foreach(@file.to_path, headers: true, skip_blanks: true) do |row|
+        # Using Google Form headers
+        email = row["Email"]
+        timestamp = Time.parse(row["Timestamp"])
+
+        # Check for matching person in organization
+        person = Person.find_by(email: email, organization: @organization)
+        next unless person
+
+        # Skip rows that have already been imported
+        previous = FormSubmission.where(person: person, csv_timestamp: timestamp)
+        next unless previous.empty?
+
+        ActiveRecord::Base.transaction do
+          form_submission = FormSubmission.create(person: person, csv_timestamp: timestamp)
+          row.each do |col|
+            # skip Email and Timestamp col as they are saved on Form Submission
+            next if col[0] == "Email" || col[0] == "Timestamp"
+
+            FormAnswer.create(person: person, form_submission: form_submission,
```

Comment:
do we need to save the person on the FormAnswer? FormAnswer belongs_to FormSubmission, which belongs_to Person, and we can set FormAnswer belongs_to Person through FormSubmission. I think i most contexts we will be calling FormAnswer from FormSubmission rather than Person, because we will want to always be considering the FormAnswer context i.e. only give me those answers from the same submission.

Reply:
Agreed. The old submitted answer had a user column so I did it as a placeholder without thinking through the associations. I'll adjust when the FormAnswer PR is complete so I get everything correct.

Reply:
Ah yes. I’ve also asked in the PR to name FormAnswer to remove user from that model association